### PR TITLE
fix(slate-react/components/content): do not use range.extend on Safari Desktop/iOS

### PR DESCRIPTION
**Pull request by zGrav**
…Content ends in whitespace, replace it with unicode no-break space

Fix for #1481

### CODE REVIEW

File: packages/slate-react/src/components/content.js [hunk 1]
```diff
@@ -206,7 +207,7 @@ class Content extends React.Component {
     native.removeAllRanges()
 
     // COMPAT: IE 11 does not support Selection.extend
-    if (native.extend) {
+    if ((native.extend && !IS_IOS) && (native.extend && !IS_SAFARI)) {
```
Comment (skogsmaskin) [inline packages/slate-react/src/components/content.js]: Maybe ``if (native.extend && !(IS_SAFARI || IS_IOS))`` for more clarity?

File: packages/slate-react/src/components/content.js [hunk 2]
```diff
@@ -206,7 +207,7 @@ class Content extends React.Component {
     native.removeAllRanges()
 
     // COMPAT: IE 11 does not support Selection.extend
```
Comment (skogsmaskin) [inline packages/slate-react/src/components/content.js]: Comment should be updated.

File: packages/slate-react/src/components/content.js (hunk 1 above)
Comment (zGrav) [reply packages/slate-react/src/components/content.js]: you are indeed correct! ;)

File: packages/slate-react/src/components/content.js (hunk 2 above)
Comment (zGrav) [reply packages/slate-react/src/components/content.js]: Done!

Comment (skogsmaskin) [reply packages/slate-react/src/components/content.js]: Great, but also keep the IE-comment about not supporting range.extend please.

**Comment by skogsmaskin** [comment]
@zGrav Can you please utdate the title and description of this PR to make a little more sense?
Also I have a feeling it is solving something the wrong way. I will have a closer look tomorrow when I have the Safari stack around.

**Comment by kinhbang89** [comment]
ANy update on this PR? I got the same error in Safari

**Comment by ianstormtaylor** [comment]
Can anyone verify this? Looking at the support tables it says that `selection.extend` is supported in Safari? Is it something specific we're doing wrong?

**Comment by ianstormtaylor** [comment]
Fixed by https://github.com/ianstormtaylor/slate/pull/3093.

**Pull request closed without merging**